### PR TITLE
Add support for IngressClass [#5232]

### PR DESCRIPTION
**Pull request by andreadecorte**
Adding support for Ingress Classes. I added them under "Cluster" section, let me know if you see them elsewhere instead (under Service?)

image
image

### CODE REVIEW

**Comment by floreks** [comment]
Our layout is based on the https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#ingressclass-v1-networking-k8s-io

IngressClass is under Service section which means it should be the same in the Dashboard.

**Comment by andreadecorte** [comment]
Thanks @floreks , moved to the Service section, updated screenshot below
Screenshot 2022-03-24 at 11 17 08

File: src/app/frontend/typings/root.api.ts [hunk 1]
```diff
@@ -405,6 +409,11 @@ export interface StorageClass extends Resource {
   parameters: StringMap[];
 }
 
+export interface IngressClass extends Resource {
+  controller: string;
+  parameters: StringMap[];
```
Comment (floreks) [inline src/app/frontend/typings/root.api.ts:414]: I don't see this field on the API side.

Comment (andreadecorte) [reply src/app/frontend/typings/root.api.ts:414]: I am flattening here https://github.com/kubernetes/dashboard/pull/6920/files#diff-f2a62a475a29db82a76ba9f4ea8915ed8a175af848d689109ecb25e118e6aa22R46 IngressClassParametersReference to a map of strings as all the contents are strings. Should I rather keep a dedicated object for that?

Comment (floreks) [reply src/app/frontend/typings/root.api.ts:414]: Ye, but this is not the `IngressClassDetails` object, it's this one https://github.com/kubernetes/dashboard/pull/6920/files#diff-a6cfbdffa7b2f43b890a274cb3a387fcda23797c8cc90e7bada690411a689d19R38-R42

Comment (andreadecorte) [reply src/app/frontend/typings/root.api.ts]: and you're of course right, field removed from there

File: src/app/frontend/resource/discovery/ingressclass/routing.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import {DISCOVERY_ROUTE} from '../routing';
+
+import {IngressClassDetailComponent} from './detail/component';
+import {IngressClassListComponent} from './list/component';
+
+const INGRESSCLASS_LIST_ROUTE: Route = {
+  path: '',
+  component: IngressClassListComponent,
+  data: {
+    breadcrumb: 'Ingress Classes',
```
Comment (floreks) [inline src/app/frontend/resource/discovery/ingressclass/routing.ts:28]: A new way to localize breadcrumbs should be used here. It was recently merged. There is now `messages.ts` file that localizes such messages.

Comment (andreadecorte) [reply src/app/frontend/resource/discovery/ingressclass/routing.ts]: thanks, updated and rebased

**Comment by andreadecorte** [comment]
did you have any chance to take a look again at this one @floreks ?

**Review by floreks — APPROVED** [review]
/lgtm
/approve

**Comment by floreks** [comment]
@andreadecorte I think we can merge. Thanks! :slightly_smiling_face:

**Comment by andreadecorte** [comment]
> @andreadecorte I think we can merge. Thanks! 🙂

thank you for the merge!

**Pull request merged**